esi/lib/nodes.py

```
@@ -265,18 +266,38 @@ def network_attach(connection, node, attach_info):
     }
 
 
-def network_detach(connection, node, port=None):
+def network_detach(
+    connection, node_name_or_uuid, port=None, port_names_or_uuids=None, all_ports=False
+):
     """Detaches a node's bare metal port from a network port
 
     :param port: The name or ID of a network port
 
-    :returns: ``True`` if the VIF was detached, otherwise ``False``
+    :returns: A list of ``(port_uuid, bool)`` tuples, where ``bool`` is ``True`` if the port was removed
+    successfully, ``False`` otherwise.
     """
 
-    node = connection.baremetal.get_node(node)
-
     if port:
-        port = connection.network.find_port(port, ignore_missing=False)
+        port_names_or_uuids = [port]
+        warnings.warn(
+            "The 'port' parameter is deprecated and will be removed in a future release.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
+    if port and port_names_or_uuids:
+        raise ValueError("do not set both port and port_names_or_uuids")
+
+    if port_names_or_uuids and all_ports:
+        raise ValueError("do not set both port_names_or_uuids and all_ports")
```

Should we also throw an error if `port` and `all_ports` then?

---

Sure. I've updated the code to do that.